jina/clients/request/__init__.py

```diff
@@ -29,8 +29,8 @@ def request_generator(
     data: GeneratorSourceType,
     request_size: int = 0,
     mode: RequestType = RequestType.INDEX,
-    mime_type: str = None,
-    queryset: Union[AcceptQueryLangType, Iterator[AcceptQueryLangType]] = None,
+    mime_type: Optional[str] = None,
+    queryset: Union[AcceptQueryLangType, Iterator[AcceptQueryLangType], None] = None,
```

Comment:
Optional is the good way to see Union with None

Reply:
Would you prefer to either have only Union types or only Optional types in the code?

Reply:
If there is Union including None, Optional is a better option.

Reply:
It is more informative about wehter None is acceptable or not (it should be done properly but we have a lot of places where Optional is not really true).

Also Optional is a useful functional construct

Reply:
So in this case is Optional[Union[..., ...]]

Reply:
I checked the documentation.
https://docs.python.org/3/library/typing.html
It says `You can use Optional[X] as a shorthand for Union[X, None].`
So, the python docs don't care which way is used.
I still think Optional would be more obvious when you look at it.